### PR TITLE
Additional generators

### PR DESCRIPTION
**Pull request by gampleman**
This adds 2 new generators:

1. JSON Decoders
2. Fuzzers from elm-test

(The fuzzer code gen definition is actually super sweet - it's like 30 lines of code)

### CODE REVIEW

File: src/Internal/Builtin/Fuzzer.elm [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        (\name -> toValueCase name ++ "Fuzzer")
+        [ CodeGenerator.customType (\_ exps -> CG.apply [ fuzz "oneOf", CG.list (List.map Tuple.second exps) ])
+        , CodeGenerator.pipeline (\c -> CG.apply [ fuzz "constant", c ]) (\m -> CG.apply [ fuzz "andMap", m ])
+        , CodeGenerator.mapN 8 (\name a bs -> CG.apply (fuzz name :: a :: bs))
+        , CodeGenerator.map (\a b -> CG.apply [ fuzz "map", a, b ])
+        , CodeGenerator.string (fuzz "string")
+        , CodeGenerator.dict (\key val -> CG.apply [ fuzz "map", CG.fqFun [ "Dict" ] "fromList", CG.apply [ fuzz "list", CG.apply [ fuzz "pair", key, val ] ] ])
+        , CodeGenerator.float (fuzz "float")
+        , CodeGenerator.tuple (\a b -> CG.apply [ fuzz "pair", a, b ])
+        , CodeGenerator.triple (\a b c -> CG.apply [ fuzz "truple", a, b, c ])
```
Comment (MartinSStewart) [inline src/Internal/Builtin/Fuzzer.elm]: `fuzz "truple"` should this be `fuzz "triple"`?

**Review by MartinSStewart — COMMENTED** [review]
Looks good! I might have spotted a typo but if that's not a mistake then I can merge this.

File: src/Internal/Builtin/Fuzzer.elm (hunk 1 above)
Comment (gampleman) [reply src/Internal/Builtin/Fuzzer.elm]: Yeah oops

**Comment by gampleman** [comment]
OK fixed that, and also fixed a completely broken implementation of `toString`... 😦

**Pull request merged**